statsmodels/genmod/tests/test_constrained.py

```diff
@@ -118,4 +119,93 @@ def init(cls):
         mod = GLM(cls.endog, cls.exog)
         mod.exog_names[:] = ['const', 'x1', 'x2', 'x3', 'x4']
         cls.res1 = mod.fit_constrained('x1=0.5', cov_type=cov_type)
-        cls.res1._results.cov_params_default = cls.res1.normalized_cov_params
+        cls.res1._results.cov_params_default = (
+            cls.res1.normalized_cov_params * cls.res1.scale)
+
+
+class ConstrainedCompareWtdMixin(ConstrainedCompareMixin):
+
+    @classmethod
+    def setup_class(cls):
+        nobs, k_exog = 100, 5
+        np.random.seed(987125)
+        x = np.random.randn(nobs, k_exog - 1)
+        x = add_constant(x)
+        cls.aweights = np.random.randint(1, 10, nobs)
+
+        y_true = x.sum(1) / 2
+        y = y_true + 2 * np.random.randn(nobs)
+        cls.endog = y
+        cls.exog = x
+        cls.idx_uc = [0, 2, 3, 4]
+        cls.idx_p_uc = np.array(cls.idx_uc)
+        cls.idx_c = [1]
+        cls.exogc = xc = x[:, cls.idx_uc]
+        mod_ols_c = WLS(y - 0.5 * x[:, 1], xc, weights=cls.aweights)
+        mod_ols_c.exog_names[:] = ['const', 'x2', 'x3', 'x4']
+        cls.mod2 = mod_ols_c
+        cls.init()
+
+
+class TestGLMWtdGaussianOffset(ConstrainedCompareWtdMixin):
+
+    @classmethod
+    def init(cls):
+        cls.res2 = cls.mod2.fit()
+        mod = GLM(cls.endog, cls.exogc,
+                  offset=0.5 * cls.exog[:, cls.idx_c].squeeze(),
+                  var_weights=cls.aweights)
+        mod.exog_names[:] = ['const', 'x2', 'x3', 'x4']
+        cls.res1 = mod.fit()
+        cls.idx_p_uc = np.arange(cls.exogc.shape[1])
+
+
+class TestGLMWtdGaussianConstrained(ConstrainedCompareWtdMixin):
+
+    @classmethod
+    def init(cls):
+        cls.res2 = cls.mod2.fit()
+        mod = GLM(cls.endog, cls.exog, var_weights=cls.aweights)
+        mod.exog_names[:] = ['const', 'x1', 'x2', 'x3', 'x4']
+        cls.res1 = mod.fit_constrained('x1=0.5')
+
+
+class TestGLMWtdGaussianConstrained1(ConstrainedCompareWtdMixin):
+
+    @classmethod
+    def init(cls):
+        cls.res2 = cls.mod2.fit()
+        mod = GLM(cls.endog, cls.exog, var_weights=cls.aweights)
+        mod.exog_names[:] = ['const', 'x1', 'x2', 'x3', 'x4']
+        cls.res1 = mod.fit_constrained('x1=0.5')
+        cls.res1._results.cov_params_default = (
+            cls.res1.normalized_cov_params * cls.res1.scale)
```

Comment:
the test that set cov_params_default are redundant after the fix.

These tests with weights are not in  4195 and will need to be kept during merge conflict resolution and rebase.